oat-sa/tao-core
# Feature/AUT-3635/Apply the Solar Design to the Item Authoring

### Related to: https://oat-sa.atlassian.net/browse/AUT-3635

Requires: 
 - [x] https://github.com/oat-sa/tao-core-ui-fe/pull/592
 - [ ] https://github.com/oat-sa/extension-tao-itemqti/pull/2505

### Summary

Apply the Solar Design to the item authoring

![image](https://github.com/oat-sa/tao-core/assets/1500098/628ee8a0-3c16-4739-85fb-4f04b314e381)


### Details

To activate the feature, you need the feature flag `FEATURE_FLAG_SOLAR_DESIGN_ENABLED`.
```sh
php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true
```

Once activated, the CSS class `solar-design` is added to the body element of the page. Then, the stylesheet automatically applies.

The Solar Design has been partially replicated in the CSS variables. For compatibility reasons, some have been simplified.

The item authoring is now following the Solar Design.

### How to test

- check out the branch: `git checkout -t origin/feature/AUT-3635/solar-design-authoring`
- activate the feature flag: `php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true`
- log in to TAO backoffice
- look at the item authoring, compare with the UX mockups
